archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/internal/filtering/TestSourceFilter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2014-2022 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.junit.internal.filtering;
+
+import com.tngtech.archunit.junit.internal.ArchUnitTestEngine;
+import com.tngtech.archunit.junit.internal.filtering.surefire.SurefireTestNameFilter;
+import org.junit.platform.engine.EngineDiscoveryRequest;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@FunctionalInterface
+public interface TestSourceFilter {
+    Logger LOG = LoggerFactory.getLogger(ArchUnitTestEngine.class);
+
+    TestSourceFilter NOOP = (descriptor -> true);
+
+    default boolean shouldRun(TestDescriptor descriptor) {
+        return descriptor.getSource()
+                .map(this::shouldRun)
+                .orElse(true);
+    }
+
+    boolean shouldRun(TestSource source);
+
+    static TestSourceFilter forRequest(EngineDiscoveryRequest discoveryRequest) {
+        try {
+            if (SurefireTestNameFilter.appliesTo(discoveryRequest)) {
+                return new SurefireTestNameFilter(discoveryRequest);
+            }
```

Comment:
Another `else if` branch could essentially be added here if we also want to introduce a filter based on custom ArchUnit properties